libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt

```diff
@@ -631,28 +667,22 @@ class MapboxRouteLineApi(
                 )
             }
 
-            val restrictedLineExpressionProvider =
-                ifNonNull(primaryRoute) { route ->
-                    if (
-                        routeLineOptions.displayRestrictedRoadSections &&
-                        MapboxRouteLineUtils.routeHasRestrictions(primaryRoute)
-                    ) {
-                        {
-                            val routeData = extractRouteRestrictionData(route)
-                            MapboxRouteLineUtils.getRestrictedLineExpression(
-                                offset,
-                                activeLegIndex,
-                                routeLineOptions
-                                    .resourceProvider
-                                    .routeLineColorResources
-                                    .restrictedRoadColor,
-                                routeData
-                            )
-                        }
-                    } else {
-                        null
+            val restrictedLineExpressionProvider = when (restrictedExpressionData.isEmpty()) {
+                true -> null
+                false -> {
+                    {
+                        MapboxRouteLineUtils.getRestrictedLineExpression(
+                            offset,
+                            activeLegIndex,
```

Comment:
When the (multi-leg) route is initially drawn the restricted section will display regardless of whether `styleInactiveRouteLegsIndependently` is true or false.  This is the case with the current release and the work in this PR hasn't changed that.  This can be seen with the `InactiveRouteStylingWithRestrictionsActivity` in the QA app. by enabling/disabling `styleInactiveRouteLegsIndependently`.

When navigation starts and `styleInactiveRouteLegsIndependently` is true the inactive leg(s) gets colored to the appropriate color including the restricted section of the inactive leg(s).  This behavior hasn't changed between the current release and this PR and can be seen using the `InactiveRouteStylingWithRestrictionsActivity` in the QA app.

Adjustments had to be made to preserve this behavior.  Until now `activeLegIndex` was only relevant if `styleInactiveRouteLegsIndependently` was true but now it's relevant for all multi-leg routes because it's needed to calculate the masking layer gradients correctly.